akka-persistence-typed/src/main/scala/akka/persistence/typed/PersistenceId.scala

```diff
@@ -125,6 +126,9 @@ object PersistenceId {
   def ofUniqueId(id: String): PersistenceId =
     new PersistenceId(id)
 
+  @InternalApi
+  private[akka] def replicated(entityId: String, replicaId: String): PersistenceId =
+    new PersistenceId(entityId + DefaultSeparator + replicaId)
```

Comment:
should also check that the replicaId doesn't contain the separator, see `apply`

Reply:
should still include `entityTypeHint` param. we can have `replicatedUniqueId` to correspond to `ofUniqueId`

Reply:
We never pull the persistence id apart so is there a reason it can't contain the separator?

Reply:
I'll add the unique one for now and we can add the entityTypeHint when doing sharding integration

Reply:
I think the separator originates from some case in Lagom where it is tearing apart the string to reconstruct the entityId. I don't think it hurts to prevent usage of the separator in replicaId.